spec/models/user_spec.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe '#name_or_email' do
+    let!(:user) { User.new(id: 1, email: 'alice@example.com', name: '') }
+    it 'return email. if name does not exist' do
+      expect(user.name_or_email).to eq 'alice@example.com'
+    end
+    it 'return name. if name exist' do
+      user.update(name: 'alice')
```

Comment:
```suggestion
      user.name = 'alice'
```

DBに保存しなくてもテストはできます。
DBに保存するならupdate!の方がいいですね